plugins/plugin-core-statistics/lib/listeners/SearchStatisticsListener.ts

```diff
@@ -56,13 +62,18 @@ export class StatisticsSearchListener<T extends Encoding>
    * @param budgetManager The budget manager
    * @param terminationManager The termination manager
    */
-  update(
+  update<T extends Encoding>(
     searchAlgorithm: SearchAlgorithm<T>,
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    subject: SearchSubject<T>,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     budgetManager: BudgetManager<T>,
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     terminationManager: TerminationManager
   ): void {
+    const searchTime = this.timing.getTimeSinceLastEvent("searchStart");
```

Comment:
This search time is already tracked inside the budget manager

Reply:
I am aware, however, the Timing object can track any time since any event. In light of the event architecture, this seemed more appropriate to me. It also felt wrong to use a budget to track the time since a particular event. What do you think?

Reply:
A new budget will be created for each target, so we won't have to keep track of the time since the last event. I think your proposed solution could also work. I understand the desire to split these as a budget is unrelated to events. However, I wonder how this would work with parallel executions when we want to implement this in the future. The budgets would still be separate but the event will be fired in an unpredictable order.

We might need to brainstorm this a bit more.